src/cli/commands/init.js

```diff
@@ -136,6 +138,9 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
         }
       } else {
         answer = (await reporter.question(question)) || def;
+        if (entry.isShorthand && entry.isShorthand(answer)) {
+          answer = 'https://github.com/' + answer;
+        }
```

Comment:
Also, this is something that should be handled outside of the regular field resolution process imo, since it's only meant to be used on the github repository and is not generic at all. Something like this after all the questions have been asked:

```js
if (pkg.repository && GitHubResolver.isVersion(pkg.repository)) {
    pkg.repository = `http://github.com/${answer}`;
}
```